### DISCUSSION TITLE
How to set the point size in pixel units instead of actual size units?

### DISCUSSION THREAD

**Discussion by WangZP10969** [vsg-dev/VulkanSceneGraph#1064] [Q&A] [answered]
I am trying to display point clouds using vsg::points. I found that the size of the points is in actual units, that is, as the object scales, the size of the points will also change. 

Is there a way to set the size of the points in pixel units? For example, always display a circle of 3 pixels in size.
![image](https://github.com/vsg-dev/VulkanSceneGraph/assets/47802547/d7214ced-a06e-4bbc-a320-e3da3b98693b)
![image](https://github.com/vsg-dev/VulkanSceneGraph/assets/47802547/99f45cc1-26c1-4233-845d-317a10843c0a)

**Comment by robertosfield** [comment, marked as answer]
The  vertex shader would need to be changed to pass a constant size pixel

I am away on a trip and replying from phone  so can't provide more details.

**Reply by WangZP10969** [reply]
Thanks a lot for the hint, I modified brick.vert to implement the feature

//float dist = max(vsg_PointSize[1], abs(eyePos.z));
//vec4 viewport = viewportData.values[0];
//gl_PointSize = viewport[3] * (vsg_PointSize[0] / dist);
gl_PointSize = vsg_PointSize[0];
